core/src/ibc/querier/mod.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+/// This module representes all accessible interface
+use crate::ibc;
+use ibc::commitment_23::types::{create_membership_proof, create_non_membership_proof};
+use ibc::commitment_23::{CommitmentPath, CommitmentState};
+use primitives::Bytes;
+use rlp::Decodable;
+
+use crate::ibc::IdentifierSlice;
+
+pub trait Nameable {
```

Comment:
The `Nameable` is used to print a debug message.
How about changing it like `DebugName`?
Then the `name()` should be changed to `debug_name`.

I found a discussion about trait name and it looks good to me.
> If the trait has a single self-explanatory method (or a set of nearly identical methods), name it after the method: Clone, Hash, Default, Into, Write, ToOwned, AsRef, Extend.

https://github.com/rust-lang/api-guidelines/issues/28#issuecomment-418509628